src/libtexture/imagecache.cpp

```diff
@@ -852,6 +860,19 @@ ImageCacheFile::read_tile(ImageCachePerThreadInfo* thread_info, int subimage,
         thread_info->m_stats.bytes_read += b;
         m_bytesread += b;
         ++m_tilesread;
+        if (id.colortransformid() > 0) {
```

Comment:
This is nice, but it feels a little scary that when a tile gets evicted, we have to do this all over again.
Sure, that's not a reason not to, but I'm just pointlessly sighing about that, even though it seems unavoidable.

Reply:
Not sure what the alternative could be. The I/O is probably as expensive than the computation. Also, remember the context: people who really care about performance should be ensuring that their textures are in the working space when they are created, not in the middle of a shader.